compiler/src/main/scala/org/scalajs/core/compiler/Compat210Component.scala

```diff
@@ -27,6 +27,14 @@ trait Compat210Component {
     def originalName: Name = sys.error("infinite loop in Compat")
 
     def isLocalToBlock: Boolean = self.isLocal
+
+    def implClass: Symbol = NoSymbol
+
+    /** The real implementation from 2.12.0-M4 is `isTrait || isInterface`,
+     *  but we use this method in places where, previously, we had only
+     *  `isInterface`.
+     */
+    def isTraitOrInterface: Boolean = self.isInterface
```

Comment:
If somebody (in the future), uses this method in PrepJSInterop (or any phase pre-mixin for that matter) and expects it to return `isTrait || isInterface` (which exists pre-2.12 IIUC), this will work in 2.12. However, in pre-2.12, this test will fail for traits that have implementations. (Btw. why can't you just implement this with `self.isInterface || self.isTrait`?)